spec-dtslint/operators/throttle-spec.ts

```diff
@@ -0,0 +1,25 @@
+import { of, timer } from 'rxjs';
+import { throttle } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const o = of(1, 2, 3).pipe(throttle(() => timer(47))); // $ExpectType Observable<number>
+});
+
+it('should infer correctly with a Promise', () => {
+  const o = of(1, 2, 3).pipe(throttle(() => new Promise<boolean>(() => {}))); // $ExpectType Observable<number>
+});
+
+it('should support a config', () => {
+  const o = of(1, 2, 3).pipe(throttle(() => timer(47), { leading: true, trailing: true })); // $ExpectType Observable<number>
+});
+
+it('should enforce types', () => {
+  const o = of(1, 2, 3).pipe(throttle()); // $ExpectError
+  const p = of(1, 2, 3).pipe(throttle(() => {})); // $ExpectError
+});
+
+it('should enforce config types', () => {
+  const o = of(1, 2, 3).pipe(throttle(() => timer(47), { x: 1 })); // $ExpectError
+  const p = of(1, 2, 3).pipe(throttle(() => timer(47), { leading: 1, trailing: 1 })); // $ExpectError
+  const q = of(1, 2, 3).pipe(throttle(() => timer(47), null)); // $ExpectError
```

Comment:
Hmm... this seems like a bug. Should be `Observable<number>`. We need to improve our checks in this operator.

Reply:
Would this mean that we would default to the default config when `null` is passed? 
I might be wrong, but I think most of the operators accepting a config parameter have the same behavior of "throwing" an error e.g. `throttleTime`

Reply:
Ah, it seems that only `throttle` and `throttleTime` has a config parameter, I'll update both operators.
Just to be sure, is this the kind of behavior we want?
> Would this mean that we would default to the default config when null is passed?

Reply:
Both [`throttle`](https://github.com/ReactiveX/rxjs/blob/6.3.2/src/internal/operators/throttle.ts#L66) and [`throttleTime`](https://github.com/ReactiveX/rxjs/blob/6.3.2/src/internal/operators/throttleTime.ts#L55) use a parameter default for the `config` object, so passing `null` should be an error. For the default to be used either the parameter would have to be omitted or `undefined` would have to be passed. That's my understanding, anyway, so I think the tests are okay.

Reply:
To accept a `null` the implementation will have to change. There is also a separate issue with some parts of the RxJS API being typed as accepting `null` parameters and others as accepting `undefined`. Both `null` and `undefined` should be acceptable.